engines/director/director.cpp

```diff
@@ -86,10 +86,11 @@ DirectorEngine::DirectorEngine(OSystem *syst, const DirectorGameDescription *gam
 	_fixStageSize = false;
 	_fixStageRect = Common::Rect();
 	_wmMode = debugChannelSet(-1, kDebugDesktop) ? wmModeDesktop : wmModeFullscreen;
+	_wmMode |= debugChannelSet(-1, kDebug32bpp) ? Graphics::kWMMode32bpp : 0;
+
 	_wmWidth = 1024;
 	_wmHeight = 768;
-
-	_wm = nullptr;
+	_wm = new Graphics::MacWindowManager(_wmMode, &_director3QuickDrawPatterns, getLanguage());
```

Comment:
And now this broke `kidsbox` quirk. We need to do something about it.

Let's create `MacWindowManager::setMode()` and that will currently contain what is at the bottom of the constructor, that is, loading the desktop if it is requested. Of course, call it right from the constructor in order to avoid code duplication.

Start this method by comparing the mode with the already set one and if the desktop flag is changed, then load or unload the relevant bitmap.